### ISSUE TITLE
Scope hoisting breaks imported third-party module

### ISSUE THREAD

**Issue by JavaScriptBach** [parcel-bundler/parcel#7996]
# 🐛 bug report

Scope hoisting sometimes produces invalid code, which results in an error like this:

`TypeError: $360c99753861bedf$export$6f093cfa640b7166 is not a function`

## 🎛 Configuration (.babelrc, package.json, cli command)

Instructions and files for repro are here: https://github.com/JavaScriptBach/test-scope-hoist

## 🤔 Expected Behavior

Scope hoisting "just works" and doesn't change the behavior of code

## 😯 Current Behavior

Scope hoisting breaks otherwise valid code and gives an error like this:

`TypeError: $360c99753861bedf$export$6f093cfa640b7166 is not a function`

## 🔦 Context

This blocks us from turning on scope hoisting in our app.

## 💻 Code Sample

https://github.com/JavaScriptBach/test-scope-hoist

```
import mammoth from "mammoth";
mammoth.convertToHtml(...);
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.4.1
| Node             | 16.14.2
| npm/Yarn         | 8.5.0
| Operating System | Linux, macOS

**Comment by MLNW** [comment]
We have the same issue when trying to use [material-table-core](https://github.com/material-table-core/core/tree/next) with pagination enabled. Our only option at this point is to disable scope-hoisting when deploying to production ...

```console
TypeError: s.isHostComponent is not a function
    children TablePagination.js:228
    R TablePagination.js:228
    React 8
    y scheduler.production.min.js:18
    React 2
    re scheduler.production.min.js:16
    onmessage scheduler.production.min.js:12
```

**Comment by F30** [comment]
Cannot provide a minimal example unfortunately, but think we are affected as well.

In our case, a method calls `super.<mangled_name>()`. However, method names of the parent class are not mangled. This appears to have gotten broken somewhere between Parcel 2.3.2 and 2.5.0.

**Comment by F30** [comment]
Whatever my problem was, it appears to be fixed in Parcel 2.6.0.

**Comment by MLNW** [comment]
I just tried it out with v2.6.0 and v2.6.2 but receive the same errors.

**Comment by mischnic** [comment]
@MLNW Please share a reproduction. This is working for me:
```jsx
import React from "react";
import MaterialTable from "@material-table/core";

const columns = [{ title: "Name", field: "name" }];
const data = Array.from({ length: 20 }).map((_, i) => ({ name: `X${i}` }));

export const App = () => <MaterialTable columns={columns} data={data} />;
```

**Comment by MLNW** [comment]
@mischnic I tried creating a reproduction with some of the dependencies we are using but I wasn't able to.

Sadly, the project we experience the issues on is closed source so I am unable to share.
But from my failed attempts of reproduction I fear it is some minute detail in our setup or configuration that leads to our specific issue.

**Comment by JavaScriptBach** [comment]
Bump; assuming this hasn't been fixed

edit: Possibly fixed by #8652, I will try to update the repro

**Comment by JavaScriptBach** [comment]
Seems it is fixed in 2.8.2
